### PR TITLE
[cli] Add CODEOWNERS support to create-plugin

### LINKED ISSUE #139: Setup CODEOWNERS for new plugins
Configure CODEOWNERS and define how external (non-Spotify team) can add parts of the repo.

### PR DESCRIPTION
**Pull request by freben**
Fixes #139 

#### :heavy_check_mark: Checklist

- [ ] All tests are passing `yarn test`
- [ ] Screenshots attached (for UI changes)
- [ ] Relevant documentation updated
- [ ] Prettier run on changed files
- [ ] Tests added for new functionality
- [ ] [CODEOWNERS](./CODEOWNERS) updated (for new stuff)
- [ ] Regression tests added for bug fixes

### CODE REVIEW

File: packages/cli/src/commands/create-plugin/createPlugin.ts [hunk 1]
```diff
@@ -343,16 +344,39 @@ const createPlugin = async () => {
...
+        }
+
+        return true;
+      },
+    },
   ];
   const answers: Answers = await inquirer.prompt(questions);
 
   const rootDir = realpathSync(process.cwd());
+  const codeownersPath = path.join(rootDir, '.github', 'CODEOWNERS');
```
Comment (Rugvip) [inline packages/cli/src/commands/create-plugin/createPlugin.ts]: We should search in all possible paths imo, `CODEOWNERS`, `docs/CODEOWNERS`, `.github/CODEOWNERS`, `.gitlab/CODEOWNERS`.

If you can be arsed, can also leave it for now 😅

File: packages/cli/src/commands/create-plugin/createPlugin.ts [hunk 2]
```diff
@@ -369,6 +393,14 @@ const createPlugin = async () => {
       addPluginToApp(rootDir, answers.id);
     }
 
+    if (ownerIds && ownerIds.length) {
```
Comment (Rugvip) [inline packages/cli/src/commands/create-plugin/createPlugin.ts]: `ownerIds?.length` :ultrafastparrot:

File: packages/cli/src/commands/create-plugin/lib/codeowners.ts [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import fs from 'fs-extra';
+
+const TEAM_ID_RE = /^@[-\w]+\/[-\w]+$/;
+const USER_ID_RE = /^@[-\w]+$/;
+const EMAIL_RE = /^(([^<>()\[\]\.,;:\s@\"]+(\.[^<>()\[\]\.,;:\s@\"]+)*)|(\".+\"))@(([^<>()[\]\.,;:\s@\"]+\.)+[^<>()[\]\.,;:\s@\"]{2,})$/i;
```
Comment (Rugvip) [inline packages/cli/src/commands/create-plugin/lib/codeowners.ts]: 🤦‍♂

File: packages/cli/src/commands/create-plugin/lib/codeowners.ts [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+  const newDeclarationLines = newDeclarationEntries.map(entry => {
+    const entryPath = entry[0] + ' '.repeat(longestOwnedPath - entry[0].length);
+    const entryOwners = entry.slice(1);
+    return [entryPath, ...entryOwners].join(' ');
+  });
+
+  const newLines = [...commentLines, '', ...newDeclarationLines, ''];
+
+  fs.writeFileSync(codeownersFilePath, newLines.join('\n'), 'utf8');
```
Comment (Rugvip) [inline packages/cli/src/commands/create-plugin/lib/codeowners.ts]: Should make these async. We need to fix the current usages of sync as it prevents the loading spinner from working

File: packages/cli/src/commands/create-plugin/createPlugin.ts (hunk 1 above)
Comment (freben) [reply packages/cli/src/commands/create-plugin/createPlugin.ts]: Ya sure, I thought I might

File: packages/cli/src/commands/create-plugin/createPlugin.ts (hunk 2 above)
Comment (freben) [reply packages/cli/src/commands/create-plugin/createPlugin.ts]: Haha, bah, less clear intentions :)

File: packages/cli/src/commands/create-plugin/lib/codeowners.ts (hunk 3 above)
Comment (freben) [reply packages/cli/src/commands/create-plugin/lib/codeowners.ts]: Wellllll, copied off the internet, gotta be perfect right

Comment (freben) [reply packages/cli/src/commands/create-plugin/lib/codeowners.ts]: Wellllll, copied off the internet, gotta be perfect right

File: packages/cli/src/commands/create-plugin/lib/codeowners.ts (hunk 4 above)
Comment (freben) [reply packages/cli/src/commands/create-plugin/lib/codeowners.ts]: Makes sense, agreed, except that this is a couple milliseconds of work :) but yeah easy fix so let's do it anyway

File: packages/cli/src/commands/create-plugin/lib/codeowners.ts (hunk 3 above)
Comment (sladyn98) [reply packages/cli/src/commands/create-plugin/lib/codeowners.ts]: Could we just catch the most common errors instead of a complex regex expression ?

**Review by Rugvip — APPROVED** [review]

**Pull request merged**